fetch/mesh_data.go

```diff
@@ -423,3 +426,24 @@ func (b *BatchError) Error() string {
 	}
 	return builder.String()
 }
+
+func (b *BatchError) Ignore() bool {
+	for hash := range b.Errors {
+		if !b.IsIgnored(hash) {
+			return false
+		}
+	}
+	return true
+}
+
+func (b *BatchError) IsIgnored(hash types.Hash32) bool {
+	err := b.Errors[hash]
+	if err == nil {
+		return false
+	}
+	nested := &BatchError{}
+	if errors.As(err, &nested) && nested.Ignore() {
+		return true
+	}
```

Comment:
I'm not sure I fully understand that logic: If a batch containing an ignored hash will trigger all parent batches to be ignored as well? Can `BatchError`s be nested arbitrarily deep?

Reply:
it depends on the type of the object. for atxs batch error can be as long as atx chain, for ballots this is limited by the tortoise window size.

the logic here, if all dependencies are "ignored" then ignore then don't try to download this object as well.